doc/specs/stdlib_hashmaps.md

```diff
@@ -0,0 +1,2104 @@
+It is an `intent(in)` argument.
+
+##### Result character
+
+The result is a scalar integer of kind `int32`.
+
+##### Result value
+
+The result is a hash code created using the `nmhash32x` algorithm.
+
+##### Note
+
+`seeded_nmhash32x_hasher` is a wrapper to the `nmhash32x_hash` of the
+module `stdlib_hash_32bit`, which supplies a fixed seed
+to the wrapped function. `nmhash32x` is an implementation of the
+`nmhash32x` hash code of James Z. M. Gao.
+This code has good, but not great, performance on long keys, poorer
+performance on short keys.
+As a result it should give fair performance for typical hash map
+applications.
+This code passes the SMHasher tests.
+
+##### Example
+
+```fortran
+    program demo_seeded_nmhash32x_hasher
+      use stdlib_hashmap_wrappers, only: &
+         seeded_nmhash32x_hasher, key_type, set
+      use iso_fortran_env, only: int32 
+      implicit none
+      integer(int8), allocatable :: array1(:)
+      integer(int32) :: hash
+      type(key_type) :: key
+      array1 = [ 5_int8, 4_int8, 3_int8, 1_int8, 10_int8, 4_int8 ]
+      call set( key, array1 )
+      hash = seeded_nmhash32x_hasher (key)
+      print *, hash
+    end program demo_seeded_nmhash32x_hasher
+```
+
+#### `seeded_water_hasher`- calculates a hash code from a key
+
+##### Status
+
+Experimental
+
+##### Description
+
+Calculates a 32 bit hash code from an input of type `key_type`.
+
+##### Syntax
+
+`code = [[stdlib_hashmap_wrappers:seeded_water_hasher]]( key )`
+
+##### Class
+
+Pure function
+
+##### Argument
+
+`key`: Shall be a scalar expression of type `key_type`.
+It is an `intent(in)` argument.
+
+##### Result character
+
+The result is a scalar integer of kind `int32`.
+
+##### Result value
+
+The result is a hash code created using the `waterhash` algorithm.
+
+##### Note
+
+`seeded_water_hasher` is a wrapper to the `water_hash` of the
+module `stdlib_hash_32bit`, which supplies a fixed seed
+to the wrapped function. `water_hash` is an implementation of the
+`waterhash` hash code of Tommy Ettinger.
+This code has excellent performance on long keys, and good performance
+on short keys.
+As a result it should give reasonable performance for typical hash
+table applications.
+This code passes the SMHasher tests.
+
+
+##### Example
+
+```fortran
+    program demo_seeded_water_hasher
+      use stdlib_hashmap_wrappers, only: &
+         seeded_water_hasher, key_type, set
+      use iso_fortran_env, only: int32 
+      implicit none
+      integer(int8), allocatable :: array1(:)
+      integer(int32) :: hash
+      type(key_type) :: key
+      array1 = [ 5_int8, 4_int8, 3_int8, 1_int8, 10_int8, 4_int8 ]
+      call set( key, array1 )
+      hash = seeded_water_hasher (key)
+      print *, hash
+    end program demo_seeded_water_hasher
+```
+
+
+#### `set` - places the data in a derived type
+
+##### Status
+
+Experimental
+
+##### Description
+
+Places the data from `value` in a `key_type` or an `other_type`.
```

Comment:
Could you provide some details on how the key is "placed" in value?

Reply:
I don't think the details are necessary for the documentation, butter your information. For a key the internal `int8` array is allocated to the proper size, then for an `int8` value the contents are copied, while for a character string value the contents are transferred. For other data, the contents are allocated with the value as the source.